test/runtests.jl

```diff
@@ -1,18 +1,33 @@
 using ADTypes
 using Test
 
+import EnzymeCore
+
 @testset "ADTypes.jl" begin
     adtype = AutoFiniteDiff()
     @test adtype isa ADTypes.AbstractADType
     @test adtype isa AutoFiniteDiff
+    @test adtype.fdtype === Val(:forward)
+    @test adtype.fdjtype === Val(:forward)
+    @test adtype.fdhtype === Val(:hcentral)
 
     adtype = AutoForwardDiff()
     @test adtype isa ADTypes.AbstractADType
-    @test adtype isa AutoForwardDiff
+    @test adtype isa AutoForwardDiff{nothing}
+
+    adtype = AutoForwardDiff(10)
```

Comment:
In contrast to all other constructors, the convenience constructor for `AutoForwardDiff` uses positional instead of keyword arguments. Was this done on purpose?

Reply:
I think not, it was carried over from previous definition in Optimization.jl as is. Feel free to fix it in this PR